src/base/bittorrent/session.cpp

```diff
@@ -1218,10 +1223,7 @@ bool Session::addTorrent(const TorrentInfo &torrentInfo, const AddTorrentParams
 bool Session::addTorrent_impl(AddTorrentData addData, const MagnetUri &magnetUri,
                               const TorrentInfo &torrentInfo, const QByteArray &fastresumeData)
 {
-    if (!addData.resumed) {
-        if (addData.savePath.isEmpty() && isASMDisabledByDefault())
-            addData.savePath = m_defaultSavePath;
-    }
+    addData.savePath = normalizeSavePath(addData.savePath, (!addData.resumed ? m_defaultSavePath : ""));
```

Comment:
Are toy sure you don't want to check for `isASMDisabledByDefault()` too?
Also there is a potential buggy case here.
If first argument to normalizeSavePath() is empty, then it will try to use the second argument. In case it comes from resumedata it will be also empty. And normalizeSavePath() will return an empty string too. Then libtorrent will try to write to working dir aka in Program Files on Windows. Causing permission errors.